src/effects/backends/builtin/compressoreffect.cpp

```diff
@@ -0,0 +1,236 @@
+    release->setName(QObject::tr("Release (ms)"));
+    release->setShortName(QObject::tr("Release"));
+    release->setDescription(
+            QObject::tr("The Release knob sets the time that determines how "
+                        "fast the compressor will recover from the gain "
+                        "reduction once the signal falls under the threshold. "
+                        "Depending on the input signal, short release times "
+                        "may introduce a 'pumping' effect and/or distortion"));
+    release->setValueScaler(EffectManifestParameter::ValueScaler::Integral);
+    release->setUnitsHint(EffectManifestParameter::UnitsHint::Millisecond);
+    release->setRange(0, 150, 1500);
+
+    EffectManifestParameterPointer gain = pManifest->addParameter();
+    gain->setId("gain");
+    gain->setName(QObject::tr("Output gain"));
+    gain->setShortName(QObject::tr("Gain"));
+    gain->setDescription(
+            QObject::tr("The Output gain knob adjusts the level of the output "
+                        "signal after the compression was applied"));
+    gain->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
+    gain->setUnitsHint(EffectManifestParameter::UnitsHint::Decibel);
+    gain->setRange(-25, 0, 25);
+
+    return pManifest;
+}
+
+CompressorGroupState::CompressorGroupState(
+        const mixxx::EngineParameters& engineParameters)
+        : EffectState(engineParameters),
+          previousStateDB(0),
+          previousMakeUpGain(0) {
+}
+
+void CompressorEffect::loadEngineEffectParameters(
+        const QMap<QString, EngineEffectParameterPointer>& parameters) {
+    m_pThreshold = parameters.value("threshold");
+    m_pRatio = parameters.value("ratio");
+    m_pKnee = parameters.value("knee");
+    m_pAttack = parameters.value("attack");
+    m_pRelease = parameters.value("release");
+    m_pGain = parameters.value("gain");
+    m_pAutoMakeUp = parameters.value("automakeup");
+    m_pClipping = parameters.value("clipping");
+}
+
+void CompressorEffect::processChannel(
+        CompressorGroupState* pState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& engineParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(enableState);
+
+    SINT numSamples = engineParameters.samplesPerBuffer();
+
+    // Compression
+    applyCompression(pState, engineParameters, pInput, pOutput);
+
+    // Auto make up
+    if (m_pAutoMakeUp->toInt() == static_cast<int>(AutoMakeUp::AutoMakeUpOn)) {
+        applyAutoMakeUp(pState, pOutput, numSamples);
+    }
+
+    // Output gain
+    CSAMPLE gainParamDB = static_cast<CSAMPLE>(m_pGain->value());
+    SampleUtil::applyGain(pOutput, db2ratio(gainParamDB), numSamples);
+
+    // Clipping
+    if (m_pClipping->toInt() == static_cast<int>(Clipping::ClippingOn)) {
+        SampleUtil::applyClamp(pOutput, numSamples);
```

Comment:
What is the use case for this? If we consider a certain loudness like 16 LUFS this clipping is way to loud. It will artificially introduce extra noise. I can imagine that the hard clipping can be an effect that can be moved along with the threshold. 
A natural sounding tanh clipper is also useful.

Reply:
I think it is piontless here as it is, because exactly that will happen when passed to the soundcard. 
So we may either remove it, or make it a useful effect. How about removing it now and postpone the work to a follow up PR?

Reply:
Oh, I didn't know about that, I thought that without such clipping these peaks may cause problems for headphones, sound speakers and so on, especially when it broadcats to radio listeners (we never knows which speakers they use and etc.).

If so, I think it's a good idea to remove it now and do it as a separate effect, for example in #6513 .